solution/index.js

```diff
@@ -0,0 +1,155 @@
+
+const LOCAL_STORAGE_KEY = "tasks"
+function localStorageInit(){
+    const json = localStorage.getItem(LOCAL_STORAGE_KEY);
+
+    if(!json){
+      localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify({
+          "to-do": [],
+          "in-progress": [],
+           "done": []
+      }))
+    }
+    loadFromLocalStorage();
+}
+
+localStorageInit()
+
+function loadFromLocalStorage(){
+    localStorageObj = JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY));
+    Object.keys(localStorageObj).forEach((sectionId) => {
+        createSectionElement(sectionId);
+        localStorageObj[sectionId].forEach((taskText)=>{
+            createNewTask(sectionId,taskText,false);
+        })
+    });
+}
+
+document.createAttribute("selectedTask");
+
+document.addEventListener("keydown",(event) => {
+    if(document.selectedTask) {
+        modifyTask(event)
+    }
+});
+
+document.getElementById("search").addEventListener("keyup",search);
+
+
+
+
+function addItemToLocalStorage(item, list){
+        const currentDataFromLocalStorage = JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY));
+        
+        const newData = {
+            ...currentDataFromLocalStorage,
+            [list]: [...currentDataFromLocalStorage[list], item]
+        }
+        
+        localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(newData))
+}
+
+function removeItemFromLocalStorage(item,list){
+    const currentDataFromLocalStorage = JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY));
+
+    const newData = {
+        ...currentDataFromLocalStorage,
+        [list]: currentDataFromLocalStorage[list].filter(listItem => listItem !== item)
+    }
+
+    localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(newData))
+
+}
+
+function createNewTask(id,taskText = null,addToLS = true){
+
+    const textInput = taskText ?? document.getElementById(`add-${id}-task`).value;
+    if(!textInput || textInput === ""){
+        throw alert("no task input")
+    }
+    const list = document.getElementsByClassName(`${id}-tasks`)[0];
+    const newTask = document.createElement("li");
+    list.replaceChildren(newTask, ...list.children);
+    newTask.classList.add("task");
+    newTask.appendChild(document.createTextNode(textInput));
+    newTask.addEventListener("dblclick",() => {
+        removeItemFromLocalStorage(newTask.textContent ,id);
+        newTask.contentEditable = true;
+        newTask.focus();
+        newTask.onblur = () => {
+            newTask.contentEditable = false;
+            newTask.textContent ? addItemToLocalStorage(newTask.textContent ,id) : newTask.remove();
+        }
+    });
+    newTask.addEventListener("mouseenter",() => mouseOverFunc(newTask));
+     
+    if(addToLS) addItemToLocalStorage(textInput, id);
+}
+
+
+function mouseOverFunc(task){
+    document.selectedTask = task;
```

Comment:
This is just a sneaky way to create a global variable.
You can find another way.